compiler/src/dotty/tools/dotc/printing/SyntaxHighlighting.scala

```diff
@@ -111,6 +111,11 @@ object SyntaxHighlighting {
             // the braces
             case tree: LambdaTypeTree =>
               tree.tparams.foreach(tparam => highlightPosition(tparam.span, TypeColor))
+            // Also special case these to ensure that each type and arg is
+            // highlighted without the braces.
+            case tree: AppliedTypeTree =>
+              highlightPosition(tree.tpt.span, TypeColor)
+              tree.args.foreach(arg => highlightPosition(arg.span, TypeColor))
```

Comment:
Won't the call to `traverseChildren` below already take care of highlighting the args and tpt? I imagine this would also give better results with nested types e.g. so that `List[Foo[Int]]` wouldn't highlight any of the brackets. If this is what we want, perhaps we can just change the `case _: TypTree` to be `case _: TypeTree`.

Reply:
Ahhhh true, good call. I made that change. It works even better. I did notice one small _regression_ in the case where you have an incomplete type like `F[_` where the `_` was colored before but now it isn't. However I think the current situation is still much better.

Reply:
<img width="487" alt="Screenshot 2022-02-19 at 19 29 08" src="https://user-images.githubusercontent.com/13974112/154814070-a64f178a-85a2-45c5-8659-ae859735c145.png">

Now it covers the deeper nested types.

Reply:
Can you add a test for the nested case?

Reply:
Sure, done.